tests/functional/ingestion/IngestionReader.js

```diff
@@ -103,112 +104,64 @@ describe('ingestion reader tests with mock', function fD() {
     let httpServer;
     let producer;
     let zkClient;
-
-    before(done => {
+    const mongoUrl =
+    `mongodb://${testConfig.queuePopulator.mongo.replicaSetHosts}` +
+    '/db?replicaSet=rs0';
+    const client = new MongoClient(mongoUrl, {});
+    let db;
```

Comment:
nit: i think it should be possible to initialize this just one, even if client is not connected...

```suggestion
    const db = client.db('metadata', { ignoreUndefined: true });
```

(may be left as is though, just sharing)